### ISSUE TITLE
CancelToken from axios is deprecated

### ISSUE THREAD

**Issue by ReFruity** [reduxjs/redux-toolkit#2147]
CancelToken from axios is [deprecated](https://axios-http.com/docs/cancellation), so I suggest updating the [documentation](https://github.com/reduxjs/redux-toolkit/blob/master/docs/api/createAsyncThunk.mdx#listening-for-abort-events) to use [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) instead.

**Comment by phryneas** [comment]
`createAsyncThunk` already supports `AbortController` natively - this example is there to show that also other mechanisms by external libraries can be used. Do you know of any other library that uses something else than `AbortController` we can use instead of `axios` then?

**Comment by ReFruity** [comment]
@phryneas No, I only use `axios` and `fetch`. But I struggle to find documentation on how to use `AbortController` with `createAsyncThunk`. I mean, we have `thunkApi` which has `signal`, but where do I find the instance of `AbortController` to call `.abort()`?

**Comment by phryneas** [comment]
You don't need any AbortController instance, you just call abort on the thunk.

```js
const dispatchedThunk = dispatch(myThunk())

dispatchedThunk.abort()
```

**Comment by johnflux** [comment]
> You don't need any AbortController instance, you just call abort on the thunk.

But that doesn't tell axios to cancel the request.  If you look at the actual network request, it's not cancelled.

**Comment by phryneas** [comment]
@johnflux if you don't pass the `signal` in some way to axios, that will do nothing, correct. That's your responsibility.

**Comment by phryneas** [comment]
You already have a signal, why do you create another?

```js
const fetchUserByIdThunk = createAsyncThunk(
  'users/fetchById',
  async (userId: string, { signal }) => {
    const response = await axios.get(`https://reqres.in/api/users/${userId}`, {
      signal,
    })
    return response.data
  }
)

```

**Comment by phryneas** [comment]
Generally, I'd welcome a docs PR for this, but if possible, please use `fetch`, not `axios` in the example. There is almost no good reason (except niche stuff like tracking file upload progress) to use `axios` nowadays and I don't really want to steer people to waste 6kb of bundle size.

**Comment by johnflux** [comment]
I had no idea that you could use the same signal.  I haven't seen that documented anywhere, but that's cool.

For the docs, how about both please?

**Comment by hamzeloudev** [comment]
A utility package for integrating file upload/download with progress tracking into your Redux Toolkit Query API slices:

[rtk-upload-utils](https://www.npmjs.com/package/rtk-upload-utils)
